lib/ts/utils/dateProvider/defaultImplementation.ts

```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+class DateProvider {
+    private clientClockDeviationInMillis: number = 0;
```

Comment:
We should try to save and re-load this from somewhere, otherwise this will have to be re-done on every refresh/full page navigation. Not 100% required, but please check if it's easily achievable

Reply:
I did this but it means we have to read from localStorage everytime we call `getClientClockSkewInMillis`. I have explained the reason in a comment.